Automattic/wp-calypso
# Site Assembler - Add pattern categories

Related to https://github.com/Automattic/wp-calypso/issues/72563 https://github.com/Automattic/wp-calypso/pull/74149 https://github.com/Automattic/wp-calypso/issues/74622

## Screenshots

|Closed state|Open state|
|---|---|
|<img width="1246" alt="Screenshot 2566-03-21 at 17 43 48" src="https://user-images.githubusercontent.com/1881481/226583599-1b8fdceb-23ac-4296-b2bb-887f2d10ef9d.png">|<img width="1246" alt="Screenshot 2566-03-21 at 17 44 04" src="https://user-images.githubusercontent.com/1881481/226583608-5f956d21-95d3-4071-86db-441050c59ed6.png">|

**Recording - Homepage > Add patterns** 

https://user-images.githubusercontent.com/1881481/226583142-8330dcda-09f1-4adb-a821-dc98d18cee7c.mov

## Proposed Changes

* Added the flag `pattern-assembler/categories`
* Updated the static patterns data (wireframe and posts patterns) to include categories
* Fetched categories from the endpoint `/block-patterns/categories`
* Show categories ordered as in the editor pattern inserter
* Click on a category to show the Pattern List Panel
* Click on the large preview to close the panel
* Updated copy to `Replace` a pattern
* Updated copy to the `Done` button to use `Save`
* Updated tooltip on pattern lists to show the pattern name 
* Pass the prop `pattern_category` in the tracks event `calypso_signup_pattern_assembler_pattern_select_click`
* Pass the prop `pattern_categories` in the tracks event `calypso_signup_pattern_assembler_pattern_select_done_click`

## Testing Instructions



* Use the Calypso live link on the first comment
* Create a new site and click "Start designing" from the bottom of the design picker 
* Choose a header and a footer
* Create your homepage 
* Add patterns 
* Open categories
* Replace, order, and delete patterns
* Verify that the track events work as before and the `pattern_category` is added
* Continue and verify the patterns are added to a Home template on your site

## Pre-merge Checklist
